ajax/entitytreesons.php

```diff
@@ -79,32 +79,43 @@
       }
    } else { // standard node
       $node_id = $_GET['node'];
-      $query   = "SELECT ent.`id`, ent.`name`, ent.`sons_cache`, count(sub_entities.id) as nb_subs
-                  FROM `glpi_entities` as ent
-                  LEFT JOIN `glpi_entities` as sub_entities
-                     ON sub_entities.entities_id = ent.id
-                  WHERE ent.`entities_id` = '$node_id'
-                  GROUP BY ent.`id`, ent.`name`, ent.`sons_cache`
-                  ORDER BY `name`";
+      $iterator = $DB->request([
+         'SELECT' => [
+            'ent.id',
+            'ent.name',
+            'ent.sons_cache',
+            new QueryExpression('COUNT('.$DB->quoteName('sub_entities.id').') AS '.$DB->quoteName('nb_subs'))
```

Comment:
i think we can use:

```
['COUNT' => 'sub_entities.id AS nb_subs']
```

See https://github.com/glpi-project/glpi/blob/9.4/bugfixes/tests/units/DBmysqlIterator.php#L272-L273

Reply:
No, this won't work:
```php
$it = $iterator->execute([
   'SELECT'    => [
      'field',
      ['COUNT' => 'field AS dafield']
   ],
   'FROM'      => 'foo'
]);
echo $it->getSql() . "\n";
//SELECT `field`, `Array` FROM `foo`
```
```